api/v2/routes_swarm.go

```diff
@@ -13,7 +13,7 @@ import (
 
 // SwarmUpload is used to upload data to ethereum swarm
 func (api *API) SwarmUpload(c *gin.Context) {
-	if api.swarm.endpoint1 == nil {
+	if api.swarmEndpoints == nil {
 		Fail(c, errors.New("must have at least one non nil swarm client"))
 		return
```

Comment:
len(...) == 0 is more dependable than testing for nil.

Also, the error message should be updated to "must have at least one ~~non nil~~ swarm client"

Reply:
good idea